### PR TITLE
kernel: Declare a useless function

### PR DESCRIPTION
**Pull request by alevkoy**
Add a poorly declared thread function for demo purposes.

Signed-off-by: Abe Levkoy <alevkoy@chromium.org>

### CODE REVIEW

**Comment by alevkoy** [comment]
This is a proof-of-concept PR to try out using fixup commits during reviews and autosquash at the end of the review. Context: https://github.com/zephyrproject-rtos/zephyr/pull/52731

People may review this, and I'll make changes in the form of pushing fixup commits. Once people like the whole thing, I'll autosquash and push again. Hopefully, it will be obvious to the reviewers that I didn't change anything else, but it will have been easier for them to review intermediate changes than it would be if I updated a whole stack of commits and force pushed.

File: include/zephyr/kernel.h [hunk 1]
```diff
@@ -5832,6 +5832,14 @@ int k_thread_runtime_stats_get(k_tid_t thread,
  */
 int k_thread_runtime_stats_all_get(k_thread_runtime_stats_t *stats);
 
+/**
```
Comment (keith-zephyr) [inline include/zephyr/kernel.h:5835]: Please fix the trailing white space.

File: include/zephyr/kernel.h [hunk 2]
```diff
@@ -5832,6 +5832,14 @@ int k_thread_runtime_stats_get(k_tid_t thread,
  */
 int k_thread_runtime_stats_all_get(k_thread_runtime_stats_t *stats);
 
+/** 
+ * @brief Store a human readable description for a thread
+ *
+ * @param thread ID of thread
+ * @return 0 on success, negative error code on failure
+ */
+void z_describe_thread(k_tid_t thread, const char *description);
```
Comment (keith-zephyr) [inline include/zephyr/kernel.h:5841]: How about creating a symmetrical API.  `z_thread_set_friendly_name` and `z_thread_get_friendly_name`.

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

**Comment by alevkoy** [comment]
@galak Some time ago, the process improvement WG discussed creating a test PR to try out autosquash during code reviews. IDK if you want to personally give this a try, but can you add someone who might be interested to this review?

File: kernel/thread.c [hunk 3]
```diff
@@ -1118,3 +1118,11 @@ int k_thread_runtime_stats_all_get(k_thread_runtime_stats_t *stats)
 
 	return 0;
 }
+
+void z_describe_thread(k_tid_t thread, const char *description)
+{
+    int description_size = 1000; /* Max size of a description */
```
Comment (stephanosio) [inline kernel/thread.c:1124]: (dummy review) `description_size` could be 128.

File: include/zephyr/kernel.h (hunk 1 above)
Comment (alevkoy) [reply include/zephyr/kernel.h]: Fixed in https://github.com/zephyrproject-rtos/zephyr-testing/pull/203/commits/86c299436ec7d2acee25dc4f9883e8981d4f9e73.

File: include/zephyr/kernel.h (hunk 2 above)
Comment (alevkoy) [reply include/zephyr/kernel.h]: Fixed in fixup commits.

File: kernel/thread.c (hunk 3 above)
Comment (alevkoy) [reply kernel/thread.c]: Fixed in https://github.com/zephyrproject-rtos/zephyr-testing/pull/203/commits/46c6b6a9fa8d5c045322cd160ea57084f2c0f172.

**Review by alevkoy — COMMENTED** [review]
Responded to review comments in fixup commits, i.e. commits that will be squashed with previous commits before merging.

I started out using `git commit --fixup` in anticipation of using `git rebase --autosquash` later. This would have produced commit messages of the form `fixup! <previous headline>`. However, I like to use that feature locally for my own purposes, and keeping track of the fixup commits I wanted to upload vs the ones I wanted to squash locally before anyone saw them became confusing.

So I used a slightly different commit message format here, and I'll manually rearrange the commits the way that `--autosquash` would have when I finally squash. That's one lesson learned so far.

**Comment by alevkoy** [comment]
@stephanosio @keith-zephyr Looking at the PR as it currently exists, how is the re-review experience?

**Comment by carlescufi** [comment]
@alevkoy this is the wrong repo. This repo is only for testing CI and GitHub actions.
Please post a draft PR to https://github.com/zephyrproject-rtos/zephyr

**Comment by alevkoy** [comment]
@carlescufi I don't intend to submit this. I just want to demo a review workflow based on fixups.

**Comment by carlescufi** [comment]
> @carlescufi I don't intend to submit this. I just want to demo a review workflow based on fixups.

Oh apologies, I realize this now.

**Comment by alevkoy** [comment]
Happy new year! Anyone want to take a look at my fixup commits?

**Comment by stephanosio** [comment]
> Happy new year! Anyone want to take a look at my fixup commits?

Could you explain how the auto-squash part is supposed to work?

It might also be helpful if you could join the Process WG meeting next week and demonstrate how the auto-squash part is supposed to work.

**Comment by alevkoy** [comment]
Each "FIXUP" commit refers to a previous commit in the PR. The fixup commits contain changes in response to comments. My hope is that this will make it easier for the people who originally requested the changes, e.g. you, to review the newly changed code.

So please review the new commits and let me know how the experience as a reviewer is.

> Could you explain how the auto-squash part is supposed to work?

Once you like how the code looks, I'll perform a rebase in which I [squash](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) the fixup commits onto their corresponding previous commits. I'll then push again. This will be a force push, because I'm rewriting existing commits instead of just appending. But there should be essentially no code changes, and reviewers should be able to quickly verify that using the "compare" button in Github.

I will then need to request review again, because the commits changed, but it should be a formality, because the reviewers already looked at all the code. You'll just be double checking that the commits are organized in a way that you like.

If you approve, then I'll be able to submit as normal. However, I will probably not actually submit this PR, because, as noted in the commit message, the actual code change is useless. It will have served its purpose to experiment with this review process.
 
> It might also be helpful if you could join the Process WG meeting next week and demonstrate how the auto-squash part is supposed to work.

I'll do that.

**Comment by mbolivar-nordic** [comment]
@alevkoy if you're going to push for this, then I would say we should at the very least be following the commit shortlog formats expected by `git rebase --autosquash` instead of this hand-rolled format.

That said I am kind of skeptical about this but I am also happy to add this to the process WG agenda for next week. Please make sure there's a corresponding issue filed with the 'Process' label in the main zephyr git repository and send me a ping on Discord or email with a link to that issue by Tuesday mid-day US pacific time and I'll make sure it's on the agenda. Thanks!

**Comment by alevkoy** [comment]
> @alevkoy if you're going to push for this, then I would say we should at the very least be following the commit shortlog formats expected by git rebase --autosquash instead of this hand-rolled format.

I started out doing that, but I found that it interfered with my own behind-the-scenes use of `--autosquash` during my development process. I could be convinced to switch back and make do.

> Please make sure there's a corresponding issue filed with the 'Process' label in the main zephyr git repository and send me a ping on Discord or email with a link to that issue by Tuesday mid-day US pacific time and I'll make sure it's on the agenda. Thanks!

[Done](https://github.com/zephyrproject-rtos/zephyr/issues/53566). I'll follow up on Discord.

**Pull request closed without merging**